I think the regression test isn't quite right:

```
        # When use_boxcox=True at model init, Box-Cox IS applied
        # (lambda ~= 0.5 for this data, not False/0)
>       assert isinstance(res.params["use_boxcox"], float), (
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            f"Expected float lambda, got {type(res.params['use_boxcox'])}"
        )
E       AssertionError: Expected float lambda, got <class 'bool'>
```

It looks like when you pass `use_boxcox=True`, `use_boxcox` will still be `True` in the model results (i.e. it won't be replaced with the fitted Box-Cox parameter). You could maybe test that `res.params['lamda'] is not None` instead?

--- a/statsmodels/tsa/holtwinters/tests/test_holtwinters.py
+++ b/statsmodels/tsa/holtwinters/tests/test_holtwinters.py
@@ -927,8 +927,8 @@ def test_use_boxcox_fit_override():
     )
     # When use_boxcox=True at model init, Box-Cox IS applied
     # (lambda ~= 0.5 for this data, not False/0)
-    assert isinstance(res.params["use_boxcox"], float), (
-        f"Expected float lambda, got {type(res.params['use_boxcox'])}"
+    assert res.params["lamda"] is not None, (
+        f"Expected fitted lambda, got {res.params["lamda"]}"
     )
 
 

